File: javascript/1010. Pairs of Songs With Total Durations Divisible by 60.js
var numPairsDivisibleBy60 = function(time) {
    let count = 0;
    let remainders = new Array(60).fill(0);
    time.forEach((duration) => {
        let remainder = duration%60;
        let compare = (remainder == 0) ? 0 : 60-remainder;
        if(remainders[compare] > 0){
            count += remainders[compare];
        }
        remainders[remainder]++;
    })
    return count;
};

File: javascript/1010. Pairs of Songs With Total Durations Divisible by 60.java
class Solution {
    public int numPairsDivisibleBy60(int[] time) {
        int count = 0;
        int[] remainders = new int[60];
        for(int duration:time){
            int remainder = duration%60;
            int comapre = (remainder == 0) ? 0 : 60-remainder;
            if(remainders[comapre] > 0){
                count += remainders[comapre];
            }
            remainders[remainder]++;
        }
        
        return count;
    }
}
